test/e2e/cli/pkg_test.go

```diff
@@ -0,0 +1,81 @@
+// SPDX-License-Identifier: BSD-3-Clause
+// Copyright (c) 2023, Unikraft GmbH and The KraftKit Authors.
+// Licensed under the BSD-3-Clause License (the "License").
+// You may not use this file except in compliance with the License.
+
+package cli_test
+
+import (
+	. "github.com/onsi/ginkgo/v2" //nolint:stylecheck
+	. "github.com/onsi/gomega"    //nolint:stylecheck
+
+	"sigs.k8s.io/kustomize/kyaml/yaml"
+
+	fcmd "kraftkit.sh/test/e2e/framework/cmd"
+	fcfg "kraftkit.sh/test/e2e/framework/config"
+	. "kraftkit.sh/test/e2e/framework/matchers" //nolint:stylecheck
+)
+
+var _ = Describe("kraft pkg", func() {
+	var cmd *fcmd.Cmd
+
+	var stdout *fcmd.IOStream
+	var stderr *fcmd.IOStream
+
+	var cfg *fcfg.Config
+
+	BeforeEach(func() {
+		stdout = fcmd.NewIOStream()
+		stderr = fcmd.NewIOStream()
+
+		cfg = fcfg.NewTempConfig()
+
+		cmd = fcmd.NewKraft(stdout, stderr, cfg.Path())
+		cmd.Args = append(cmd.Args, "pkg")
+	})
+
+	var _ = Describe("update", func() {
+		var manifestsPath string
+
+		BeforeEach(func() {
+			cmd.Args = append(cmd.Args, "update")
+
+			manifestsPath = yaml.GetValue(cfg.Read("paths", "manifests"))
+			Expect(manifestsPath).To(SatisfyAny(
+				Not(BeAnExistingFile()),
+				BeAnEmptyDirectory(),
+			), "manifests directory should either be empty or not yet created")
+		})
+
+		When("invoked without flags or positional arguments", func() {
+			It("should retrieve the list of components, libraries and packages", func() {
+				err := cmd.Run()
+				Expect(err).ToNot(HaveOccurred())
+
+				Expect(stderr.String()).To(BeEmpty())
+				// The commands sends control characters while updating.
+				Expect(stdout.String()).To(MatchRegexp(`\[\+\].+Updating\.\.\. \[\d+\.\d+s\]\r\n`))
+
+				Expect(manifestsPath).To(ContainFiles("index.yaml", "unikraft.yaml"))
+				Expect(manifestsPath).To(ContainDirectories("libs"))
```

Comment:
These checks are manifest-specific artifacts. Whilst the default functionality of kraftkit is to use manifest Package Manager, it can be configured to use a different ones which would result in alternative output artefacts from `kraft pkg update`. For example, after the introduction of OCI package manager, sourcing from `unikraft.org` does not actually produce any system artefacts.

Perhaps just for greater clarity of the test(s) itself the package manager should be indicated? The flag `-m` should then be used to specifically set the manifest manager and the test name also includes this. 

I can then see additional tests that [try out different providers](https://github.com/unikraft/kraftkit/blob/staging/manifest/provider.go#L34-L91), e.g., starting a blank list of unikraft manifests:
```
# e2e test 1: GitHub provider
kraft pkg source https://github.com/unikraft/unikraft.git
kraft pkg update
# only a `unikraft.yaml` artefact

# e2e test 2: GitProvider
[...]

# e2e test N: ManifestIndexProvider
kraft pkg source https://manifests.kraftkit.sh/index.yaml
# `index.yaml` and any additional "symlinks"
[...]
```

This functionality assumes (afaik) that the config will be populated with its default configuration and values, meaning that the manifest provider is an [ManifestIndexProvider](https://github.com/unikraft/kraftkit/blob/staging/manifest/index.go#L33) set to [the default location](https://github.com/unikraft/kraftkit/blob/staging/config/defaults.go#L17).

Reply:
How about having both? Test with a default config, and test with selected manifests?

Reply:
I just nested the existing test cases inside a context block (see [eedbbb1f..aa9bfc3b](https://github.com/unikraft/kraftkit/compare/eedbbb1f2bc921725d4822cd042166da9efc2bd0..aa9bfc3bbb65259a98466662ab7d62518456aafb)):

```go
Context("implicitly using the default manager type (manifest)", { /* ... */ })
```

This should make it explicit that we are testing with plain defaults, and that future context blocks will contain manager-specific test cases, alongside the default case.

PTAL

Reply:
You're suggestion of updating the test context helps assure that this test is intended for "default" functionality and addresses my concern. Thanks a tonne!

I will open up a separate issue addressing the permutations that can be made by the `--manager` flag which is out-of-scope for this initial e2e of `kraft pkg update`.